dstep/translator/Output.d

```diff
@@ -14,124 +14,287 @@ import mambo.core._;
 
 import clang.Cursor;
 
-import dstep.translator.CodeBlock;
+import dstep.translator.Context;
 import dstep.translator.IncludeHandler;
 import dstep.translator.Type;
 
-Output output;
-
-static this ()
+class Output
 {
-    output = new Output;
-}
+    this()
+    {
+        stack ~= Entity.bottom;
+    }
 
-void resetOutput()
-{
-    output = new Output;
-}
+    public bool empty()
+    { 
+        return stack.length == 1 && 
+            stack.back == Entity.bottom &&
+            weak == null;
+    }
 
-class Output
-{
-    String currentContext;
+    public void separator()
+    {
+        if (stack.back == Entity.singleLine)
+            stack.back = Entity.separator;
+    }
 
-    alias currentContext this;
+    public void output(Output output)
+    {
+        if (output.stack == [Entity.singleLine])
```

Comment:
I think it's better to check the length of `stack` and compare the first element than allocating a new array to compare with.